packages/types/src/payment/service.ts

```diff
@@ -1,3 +1,121 @@
 import { IModuleService } from "../modules-sdk"
+import { Context } from "../shared-context"
+import {
+  CreatePaymentCollectionDTO,
+  CreatePaymentDTO,
+  CreatePaymentSessionDTO,
+  SetPaymentSessionsDTO,
+  UpdatePaymentCollectionDTO,
+  UpdatePaymentDTO,
+} from "./mutations"
+import {
+  FilterablePaymentCollectionProps,
+  PaymentCollectionDTO,
+  PaymentDTO,
+} from "./common"
+import { FindConfig } from "../common"
 
-export interface IPaymentModuleService extends IModuleService {}
+export interface IPaymentModuleService extends IModuleService {
+  /* ********** PAYMENT COLLECTION ********** */
+
+  createPaymentCollection(
+    data: CreatePaymentCollectionDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO[]>
+  createPaymentCollection(
+    data: CreatePaymentCollectionDTO,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  retrievePaymentCollection(
+    paymentCollectionId: string,
+    config?: FindConfig<PaymentCollectionDTO>,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  listPaymentCollections(
+    filters?: FilterablePaymentCollectionProps,
+    config?: FindConfig<PaymentCollectionDTO>,
+    sharedContext?: Context
+  ): Promise<[PaymentCollectionDTO[], number]>
+
+  listAndCountPaymentCollections(
+    filters?: FilterablePaymentCollectionProps,
+    config?: FindConfig<PaymentCollectionDTO>,
+    sharedContext?: Context
+  ): Promise<[PaymentCollectionDTO[], number]>
+
+  updatePaymentCollection(
+    data: UpdatePaymentCollectionDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO[]>
+  updatePaymentCollection(
+    data: UpdatePaymentCollectionDTO,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  deletePaymentCollection(
+    paymentCollectionId: string[],
+    sharedContext?: Context
+  ): Promise<void>
+  deletePaymentCollection(
+    paymentCollectionId: string,
+    sharedContext?: Context
+  ): Promise<void>
+
+  authorizePaymentCollection(
+    paymentCollectionId: string,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  completePaymentCollection(
+    paymentCollectionId: string,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  /* ********** PAYMENT ********** */
+
+  createPayment(data: CreatePaymentDTO): Promise<PaymentDTO>
+
+  capturePayment(
+    paymentId: string,
+    amount: number,
+    sharedContext?: Context
+  ): Promise<PaymentDTO>
+  refundPayment(
+    paymentId: string,
+    amount: number,
+    sharedContext?: Context
+  ): Promise<PaymentDTO>
+
+  updatePayment(
+    data: UpdatePaymentDTO,
+    sharedContext?: Context
+  ): Promise<PaymentDTO>
+
+  /* ********** PAYMENT SESSION ********** */
+
+  createPaymentSession(
+    paymentCollectionId: string,
+    data: CreatePaymentSessionDTO,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
```

Comment:
todo: I think we should be able to do it in bulk too

Reply:
As in multiple payment collections and not just multiple sessions for a single collection?

Reply:
Only the latter is currently supported

Reply:
Same goes for the rest of the methods